test/jasmine/tests/frame_api_test.js

```diff
@@ -0,0 +1,205 @@
+var Plotly = require('@lib/index');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+
+// A helper function so that failed tests don't simply stall:
+function fail(done) {
```

Comment:
Was gonna ask about a common place for these helpers. Unfortunately, `err.stack` doesn't seem to follow the source map, so it's still not totally straightforward to track it down. But it *does* prevent failed promise-based tests from hanging, which is pretty lovely!